File: src/tui/theme.ts
import { KNOWN_PROVIDERS } from "../registry.ts";
import type { UsageStatus } from "../usage/types.ts";
import { type HeatLevel, heatLevel } from "../report/graph.ts";

/** Ink color name per usage status, mirroring the CLI breakdown palette. */
export const statusColor: Record<UsageStatus, string> = {
  exhausted: "red",
  warning: "yellow",
  ok: "green",
  unknown: "gray",
};

/** Ink color name per heat band, matching the CLI graph's ANSI ramp. */
const HEAT_INK: Record<HeatLevel, string> = {
  hot: "red",
  busy: "yellow",
  calm: "green",
  idle: "gray",
};

export function heatInk(ratio: number): string {
  return HEAT_INK[heatLevel(ratio)];
}

/**
 * Per-provider color band (ink/chalk names) cycled so each row/chart is
 * visually distinct. Indexed by the provider's position in KNOWN_PROVIDERS so
 * the color is stable across runs. Port of `theme.rs::PALETTE`.
 */
export const PALETTE: string[] = [
  "cyan",
  "green",
  "yellow",
  "magenta",
  "blue",
  "redBright",
  "cyanBright",
  "greenBright",
  "yellowBright",
  "magentaBright",
  "blueBright",
  "red",
];

export function colorFor(id: string): string {
  let idx = KNOWN_PROVIDERS.findIndex((p) => p.id === id);
  if (idx < 0) {
    let sum = 0;
    for (const b of Buffer.from(id, "utf8")) sum += b;
    idx = sum;
  }
  return PALETTE[idx % PALETTE.length]!;
}

File: legacy-rust/src/tui/theme.rs
use ratatui::style::{Color, Modifier, Style};

/// Color band for usage gauges. <70 green, 70..90 yellow, >=90 red.
pub fn gauge_color(pct: f64) -> Color {
    if pct >= 90.0 { Color::Red }
    else if pct >= 70.0 { Color::Yellow }
    else { Color::Green }
}

pub const SELECTED: Style = Style::new().add_modifier(Modifier::BOLD).fg(Color::Cyan);
pub const HEADER: Style = Style::new().add_modifier(Modifier::BOLD);
pub const MUTED: Color = Color::DarkGray;

/// btop-style color band cycled per provider so each row/chart is visually
/// distinct. Indexed by the provider's position in `KNOWN_PROVIDERS` so the
/// color is stable across runs.
pub const PALETTE: &[Color] = &[
    Color::Cyan,
    Color::Green,
    Color::Yellow,
    Color::Magenta,
    Color::Blue,
    Color::LightRed,
    Color::LightCyan,
    Color::LightGreen,
    Color::LightYellow,
    Color::LightMagenta,
    Color::LightBlue,
    Color::Red,
];

pub fn color_for(id: &str) -> Color {
    let idx = crate::registry::KNOWN_PROVIDERS
        .iter()
        .position(|p| p.id == id)
        .unwrap_or_else(|| id.bytes().map(|b| b as usize).sum());
    PALETTE[idx % PALETTE.len()]
}